test/libsolidity/SemanticTest.cpp

```diff
@@ -244,15 +268,27 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
 		for (auto const& test: m_tests)
 		{
 			ErrorReporter errorReporter;
-			_stream << test.format(errorReporter, _linePrefix, false, _formatted) << endl;
+			_stream << test.format(
+					errorReporter,
+					_linePrefix,
+					TestFunctionCall::RenderMode::ExpectedValuesExpectedGas,
+					_formatted,
+					/* _interactivePrint */ true
+				) << endl;
 			_stream << errorReporter.format(_linePrefix, _formatted);
 		}
 		_stream << endl;
 		AnsiColorized(_stream, _formatted, {BOLD, CYAN}) << _linePrefix << "Obtained result:" << endl;
 		for (auto const& test: m_tests)
 		{
 			ErrorReporter errorReporter;
-			_stream << test.format(errorReporter, _linePrefix, true, _formatted) << endl;
+			_stream << test.format(
+					errorReporter,
+					_linePrefix,
+					m_gasCostFailure ? TestFunctionCall::RenderMode::ExpectedValuesActualGas : TestFunctionCall::RenderMode::ActualValuesExpectedGas,
```

Comment:
I was thinking to do like this, but there were some tests that were passing checks, but when updating it updates the test when it really shouldn't. Mostly because of encoding. I would prefer to finish up this PR without tackling this one now, in order to have gas costs as soon as possible, then would add this change in separate PR if that is fine with you?

Reply:
Yeah, sounds fine. It might be a potential future problem but the code works as it is now so it's not really a high priority.